django_cloud_deploy/cli/prompt.py

```diff
@@ -838,6 +860,205 @@ def _validate(self, s):
+                   'database: ').format(current_step, total_steps)
+            host = _ask_prompt(msg, console)
+            default_port = 5432
+            msg = ('[{}.b/{}] Enter the port number of your database or '
+                   'press Enter to use "{}":').format(current_step, total_steps,
+                                                      default_port)
+            port = _ask_prompt(msg, console, default=default_port)
+            msg = ('[{}.c/{}] Enter the master user name for the '
+                   'database: ').format(current_step, total_steps)
+            username = _ask_prompt(msg, console, _database_username_validate)
+            msg = '[{}.d/{}] Enter password for the database user "{}"'.format(
+                current_step, total_steps, username)
+            password = _password_prompt(msg, console)
+            msg = '[{}.e/{}] Enter the name of your database: '.format(
+                current_step, total_steps)
+            database_name = _ask_prompt(msg, console)
+            new_args['is_new_database'] = False
+            new_args['database_username'] = username
+            new_args['database_password'] = password
+            new_args['database_host'] = host
+            new_args['database_port'] = int(port)
+            new_args['database_name'] = database_name
+            try:
+                self._validate(new_args)
+                break
+            except ValueError as e:
+                console.error(e)
+
+        return new_args
+
+    def _try_get_database_arguments(self, args: Dict[str, Any]
+                                   ) -> Optional[Dict[str, Any]]:
+        try:
+            return {
+                'host': args['database_host'],
+                'port': args['database_port'],
+                'user': args['database_username'],
+                'password': args['database_password'],
+                'database': args['database_name'],
+            }
+        except KeyError:
+            return None
+
+    def _validate(self, database_arguments: Dict[str, Any]):
+        """Validate the given database info by connecting to it.
+
+        Args:
+            database_arguments: Dictionary holding the information of database.
+
+        Raises:
+            ValueError: If failed to connect to the provided database.
+        """
+
+        host = database_arguments.get('database_host')
+        port = database_arguments.get('database_port')
+        user = database_arguments.get('database_username')
+        password = database_arguments.get('database_password')
+        database = database_arguments.get('database_name')
+        conn = None
+        try:
+            conn = psycopg2.connect(host=host,
+                                    database=database,
+                                    user=user,
+                                    password=password,
+                                    port=port)
+        except psycopg2.Error as e:
+            raise ValueError(
```

Comment:
Is it worth logging the values the user provided to help with debugging? (WIthout the password)

Reply:
The values will be shown to the user when they fill in the values:

```
[2/12] Do you want to create a new database or use an existing database for deployment? [y/N]: y
[2.a/12] Enter the public ip or host name of your database: mydbinstance3.ckukagg5823x.us-west-2.rds.amazonaws.com
[2.b/12] Enter the port number of your database or press Enter to use "5432":
[2.c/12] Enter the master user name for the database: admin
[2.d/12] Enter password for the database user "admin"
Password: 
Password (again): 
[2.e/12] Enter the name of your database: mydb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/google/home/wangxf/Desktop/code/django-cloud-deploy/venv/lib/python3.6/site-packages/psycopg2/__init__.py", line 130, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: FATAL:  password authentication failed for user "admin"
FATAL:  password authentication failed for user "admin"
```